servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java

```diff
@@ -27,7 +28,7 @@
  * A greedy allocator that returns the first Storage dir fitting the size of block to allocate.
  * This class serves as an example how to implement an allocator.
  */
-public class GreedyAllocator implements Allocator {
+public class GreedyAllocator extends BlockStoreEventListenerBase implements Allocator {
```

Comment:
@miasantreble Originally I thought abstract class would be good, but now I think there's value in keeping it an interface. Do you have a strong opinion either way? If we go with interface, I think the ticket is no longer necessary.